crates/ln-dlc-node/src/tests/mod.rs

```diff
@@ -1,6 +1,368 @@
+        .channel_manager
+        .list_channels()
+        .get(index)
+        .unwrap()
+        .clone();
+
+    let channel_id = hex::encode(details.channel_id);
+    let short_channel_id = details.short_channel_id.unwrap();
+    let is_ready = details.is_channel_ready;
+    let is_usable = details.is_usable;
+    let inbound = details.inbound_capacity_msat;
+    let outbound = details.outbound_capacity_msat;
+    tracing::info!(
+        channel_id,
+        short_channel_id,
+        is_ready,
+        is_usable,
+        inbound,
+        outbound,
+        "{pair}"
+    );
+}
+
+async fn wait_until<P, T, F>(timeout: Duration, predicate_fn: P) -> Result<T>
+where
+    P: Fn() -> F,
+    F: Future<Output = Result<Option<T>>>,
+{
+    tokio::time::timeout(timeout, async {
+        loop {
+            match predicate_fn().await? {
+                Some(value) => return Ok(value),
+                None => tokio::time::sleep(Duration::from_millis(500)).await,
+            };
+        }
+    })
+    .await?
+}
+
+fn dummy_contract_input(
+    offer_collateral: u64,
+    accept_collateral: u64,
+    oracle_pk: XOnlyPublicKey,
+) -> ContractInput {
+    let total_collateral = offer_collateral + accept_collateral;
 
-#[cfg(feature = "system_tests")]
-mod system_tests;
+    ContractInput {
+        offer_collateral,
+        accept_collateral,
+        maturity_time: 1653378780,
+        fee_rate: 2,
+        contract_infos: vec![ContractInputInfo {
+            contract_descriptor: ContractDescriptor::Numerical(NumericalDescriptor {
+                payout_function: PayoutFunction {
+                    payout_function_pieces: vec![
+                        PayoutFunctionPiece::PolynomialPayoutCurvePiece(
+                            PolynomialPayoutCurvePiece::new(vec![
+                                PayoutPoint {
+                                    event_outcome: 0,
+                                    outcome_payout: 0,
+                                    extra_precision: 0,
+                                },
+                                PayoutPoint {
+                                    event_outcome: 50_000,
+                                    outcome_payout: 0,
+                                    extra_precision: 0,
+                                },
+                            ])
+                            .unwrap(),
+                        ),
+                        PayoutFunctionPiece::PolynomialPayoutCurvePiece(
+                            PolynomialPayoutCurvePiece::new(vec![
+                                PayoutPoint {
+                                    event_outcome: 50_000,
+                                    outcome_payout: 0,
+                                    extra_precision: 0,
+                                },
+                                PayoutPoint {
+                                    event_outcome: 60_000,
+                                    outcome_payout: total_collateral,
```

Comment:
Is this code somehow creating the curve?

Reply:
I think this code _is_ the piecewise polynomial curve. Later, `rust-dlc` uses this to interpolate the values for intermediate payouts.

One thing we need to figure out is how to configure this to generate a set number of payouts. Because atm the test generates 7.5k CETs.

Reply:
> One thing we need to figure out is how to configure this to generate a set number of payouts. Because atm the test generates 7.5k CETs.

ufff...